backend/src/s9pk/mod.rs

```diff
@@ -122,23 +123,52 @@ pub async fn pack(#[context] ctx: SdkContext, #[arg] path: Option<PathBuf>) -> R
             std::io::Cursor::new(assets.into_inner().await?)
         })
         .scripts({
-            let script_path = path.join(manifest.assets.scripts_path()).join("embassy.js");
+            let script_base_path = path.join(manifest.assets.scripts_path());
             let needs_script = manifest.package_procedures().any(|a| a.is_script());
-            let has_script = script_path.exists();
-            match (needs_script, has_script) {
-                (true, true) => Some(File::open(script_path).await?),
-                (true, false) => {
-                    return Err(Error::new(eyre!("Script is declared in manifest, but no such script exists at ./scripts/embassy.js"), ErrorKind::Pack).into())
-                }
-                (false, true) => {
+            let script_path_js = script_base_path.join("embassy.js");
+            let script_path_ts = script_base_path.join("embassy.ts");
+            match (needs_script, script_path_js.exists(), script_path_ts.exists()) {
+                (false, false, false) => None,
+                (false, true, _) => {
                     tracing::warn!("Manifest does not declare any actions that use scripts, but a script exists at ./scripts/embassy.js");
                     None
                 }
-                (false, false) => None
+                (false, false, true) => {
+                    tracing::warn!("Manifest does not declare any actions that use scripts, but a script exists at ./scripts/embassy.ts");
+                    None
+                }
+                (true, false, false) => {
+                    return Err(Error::new(eyre!("Script is declared in manifest, but no such script exists at ./scripts/embassy.js"), ErrorKind::Pack))
+                }
+                (true, true, true) => {
+                    tracing::warn!("Both ./scripts/embassy.js and ./scripts/embassy.ts exist. Using embassy.js, skipping Typescript bundling...");
```

Comment:
```suggestion
```
I'm 100% against the warning while the ts bundling in not working/ validating.

Reply:
Are you saying you want to disable the embassy.ts? I think giving a warning that tells the user how the thing is choosing to disambiguate a request is a very important idea. I would hazard a guess that your beef isn't with the warning but about the actual behavior. Can you describe what you'd like to be different about ti?

Reply:
Right now, with no checking, I feel that it would be better to bundle via `deno bundle` or `webpack pack` or `rollup`
It would be better because then they are getting the bundler's ability to lint and typecheck.